plugin/evm/gossip_mempool.go

```diff
@@ -0,0 +1,217 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package evm
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/ava-labs/coreth/gossip"
+	"github.com/ava-labs/coreth/plugin/evm/message"
+
+	bloomfilter "github.com/holiman/bloomfilter/v2"
+
+	"github.com/ava-labs/coreth/core"
+	"github.com/ava-labs/coreth/core/txpool"
+	"github.com/ava-labs/coreth/core/types"
+)
+
+var (
+	_ gossip.Set[*GossipAtomicTx] = (*GossipAtomicMempool)(nil)
+	_ gossip.Gossipable           = (*GossipAtomicTx)(nil)
+
+	_ gossip.Set[*GossipEthTx] = (*GossipEthTxPool)(nil)
+	_ gossip.Gossipable        = (*GossipEthTx)(nil)
+)
+
+func NewGossipAtomicMempool(Mempool *Mempool) (*GossipAtomicMempool, error) {
```

Comment:
Why wrap the atomic mempool rather than directly implementing the required interface with the original atomic mempool?

This seems like an extra layer of indirection.

Makes sense for the eth tx pool, so that we don't need to add our changes to the geth implementation, but wondering if we do it here for symmetry or what the rationale is for wrapping the atomic mempool as opposed to implementing that interface directly in this case

Reply:
Yeah this was done for consistency. I'll see if we can just implement this directly into the atomic mempool.